tests/unit/dataactvalidator/test_a20_appropriations_1.py

```diff
@@ -0,0 +1,32 @@
+from tests.unit.dataactcore.factories.staging import AppropriationFactory
+from tests.unit.dataactcore.factories.domain import CGACFactory
+from tests.unit.dataactvalidator.utils import number_of_errors, query_columns
+from random import choice
+from string import ascii_uppercase, ascii_lowercase, digits
+
+_FILE = 'a20_appropriations_1'
+
+def test_column_headers(database):
+    expected_subset = {"row_number", "allocation_transfer_agency"}
+    actual = set(query_columns(_FILE, database))
+    assert expected_subset <= actual
+
+def test_success(database):
+    """ Test that agency codes are matched against cgac correctly """
+    approp = AppropriationFactory()
+    approp_null = AppropriationFactory(allocation_transfer_agency = None)
```

Comment:
The comparison between approp and cgac is for a non-null allocation_transfer_agency, the AppropriationFactory generates a random string and cgac reuses that string.  Is that the case you're referring to or is there another case I'm not thinking of here?